puppet/modules/jenkins_job_builder/files/centos.org/pipelines/foreman-discovery-image-build.groovy

```diff
@@ -0,0 +1,54 @@
+pipeline {
+    agent { label 'foreman' }
+
+    environment {
+        proxy_repository = env.getProperty('proxy_repository')
+        repoowner = env.getProperty('repoowner')
+        branch = env.getProperty('branch')
+    }
+
+    stages {
+        stage('Setup Environment') {
+            steps {
+                deleteDir()
+                git url: 'https://github.com/theforeman/forklift.git'
+            }
+        }
+        stage('Provision Node') {
+            steps {
+                provision()
+            }
+        }
+        stage('Install Pipeline Requirements') {
+            steps {
+                runPlaybook(
+                    playbook: 'playbooks/setup_forklift.yml',
+                    inventory: cico_inventory('./'),
+                )
+            }
+        }
+        stage('Run Build') {
+            steps {
+                script {
+                    duffy_ssh("git clone https://github.com/${env.repoowner}/foreman-discovery-image/ --branch ${env.branch}", 'duffy_box', './')
```

Comment:
Do we ever want to build it not from the official source in our infra? Normally I'm all for configuration but in this case it's probably not desired (unless we go all the way and make it `env.git_url`).

Reply:
dropped the repoowner param after talking with lzap